pkg/volume/util.go

```diff
@@ -129,28 +160,63 @@ func (c *realRecyclerClient) DeletePod(name, namespace string) error {
 	return c.client.Core().Pods(namespace).Delete(name, nil)
 }
 
+func (c *realRecyclerClient) Event(eventtype, message string) {
+	c.recorder(eventtype, message)
+}
+
 // WatchPod returns a ListWatch for watching a pod.  The stopChannel is used
 // to close the reflector backing the watch.  The caller is responsible for
 // derring a close on the channel to stop the reflector.
-func (c *realRecyclerClient) WatchPod(name, namespace string, stopChannel chan struct{}) func() *api.Pod {
-	fieldSelector, _ := fields.ParseSelector("metadata.name=" + name)
-
-	podLW := &cache.ListWatch{
-		ListFunc: func(options api.ListOptions) (runtime.Object, error) {
-			options.FieldSelector = fieldSelector
-			return c.client.Core().Pods(namespace).List(options)
-		},
-		WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
-			options.FieldSelector = fieldSelector
-			return c.client.Core().Pods(namespace).Watch(options)
-		},
+func (c *realRecyclerClient) WatchPod(name, namespace string, stopChannel chan struct{}) (<-chan watch.Event, error) {
+	podSelector, _ := fields.ParseSelector("metadata.name=" + name)
+	options := api.ListOptions{
+		FieldSelector: podSelector,
+		Watch:         true,
```

Comment:
off-topic, it seems redundant that you should have to specify `Watch=true` for the watch API endpoint